### DISCUSSION TITLE
Error: EROFS: read-only file system using tmp folder

### DISCUSSION THREAD

**Discussion by DVGY** [vercel/community#5812] [Help] [answered]
### Page to Investigate

https://dashboard-showcase-staging-vx21vsukh-techispapa.vercel.app/projects?_sort=name&_order=null

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/46158826/e06d67b5-c164-4a94-b7da-abf4376c6657).

**Comment by amyegan** [comment, marked as answer]
Hi @DVGY. It looks like your app is trying to write something to a location that is read-only. We have a guide covering use of files in Serverless Functions that I recommend. You can find it at https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage

**Reply by DVGY** [reply]
I have used tmp folder, still it is throwing error

**Comment by DVGY** [comment]
@amyegan Can you please help me here, it is still not working

**Reply by amyegan** [reply]
The error suggests that something in the app is trying to write to a read-only location. I recommend reviewing the [temporary storage guide](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage) and double checking the app code with those guidelines. 

If you need more help, please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so people here can see how the code is set up for this function. That way someone may be able to review it, recognize the problem, and suggest a specific solution.
